File: javascript/src/models/s3Config.ts
// this file is @generated

export interface S3Config {
  bucket: string;
  accessKeyId: string;
  secretAccessKey: string;
  region: string;
  endpointUrl?: string | null;
}

export const S3ConfigSerializer = {
  _fromJsonObject(object: any): S3Config {
    return {
      bucket: object["bucket"],
      accessKeyId: object["accessKeyId"],
      secretAccessKey: object["secretAccessKey"],
      region: object["region"],
      endpointUrl: object["endpointUrl"],
    };
  },

  _toJsonObject(self: S3Config): any {
    return {
      bucket: self.bucket,
      accessKeyId: self.accessKeyId,
      secretAccessKey: self.secretAccessKey,
      region: self.region,
      endpointUrl: self.endpointUrl,
    };
  },
};

File: kotlin/lib/src/main/kotlin/models/S3Config.kt
// This file is @generated
package com.svix.kotlin.models

import kotlinx.serialization.Serializable

@Serializable
data class S3Config(
    val bucket: String,
    val accessKeyId: String,
    val secretAccessKey: String,
    val region: String,
    val endpointUrl: String? = null,
)
